micro_dl/utils/masks.py

```diff
@@ -10,19 +10,38 @@
 from micro_dl.utils.image_utils import im_adjust
 
 
-def create_otsu_mask(input_image, kernel_size=3):
+def create_otsu_mask(input_image, style='otsu', thresh_input=0, kernel_size=11):
     """Create a binary mask using morphological operations
 
     :param np.array input_image: generate masks from this image
-    :param int str_elem_size: size of the structuring element. typically 3, 5
+    :param str style: 'otsu', or 'binary'
+    :param Iterable thresh_input: default zero, chnages size to input max and min intensity & otsu threshold level of stack middle slice.
+    :param int kernel_size: Gaussian blur kernel size. odd number and increase with increase in size of object
     :return: mask of input_image, np.array
     """
 
-    input_image = im_adjust(
-        cv2.GaussianBlur(input_image, (kernel_size, kernel_size), 0)
-    )
-    _, mask = cv2.threshold(input_image, 0, 255, cv2.THRESH_BINARY + cv2.THRESH_OTSU)
-    return mask
+    ret = []
+    input_image = input_image.astype("float32")
+    input_image_blur = cv2.GaussianBlur(input_image, (kernel_size, kernel_size), 0)
+    
+    # if middle slice
+    if style == 'otsu':
+        focus_max = np.max(input_image_blur)
+        ret.append(focus_max)
+        focus_min = np.min(input_image_blur)
+        ret.append(focus_min)
+        input_image_norm = 255*(input_image_blur - focus_min)/(focus_max - focus_min)
+
+    
+        thresh, mask = cv2.threshold(input_image_norm.astype(np.uint8), 0, 255, cv2.THRESH_BINARY + cv2.THRESH_OTSU)
+        ret.append(thresh)
+
+    # for the whole stack based on values computed from middle stack
+    elif style == 'binary':
+        input_image_norm = 255*(input_image_blur - thresh_input[1])/(thresh_input[0] - thresh_input[1])
+        ret, mask = cv2.threshold(input_image_norm.astype(np.uint8), thresh_input[2], 255, cv2.THRESH_BINARY)
+    
+    return mask, ret
```

Comment:
What is `ret` here? The docstring says that only a mask is returned.

Reply:
Sorry @ziw-liu , I forgot to add that. I have added it now.